src/System.CommandLine.Tests/Binding/ModelBinderConstructorTests.cs

```diff
@@ -0,0 +1,140 @@
+
+namespace System.CommandLine.Tests.Binding
+{
+    public class ModelBinderConstructorTests
+    {
+
+        [Theory]
+        [InlineData(typeof(ClassWithCtorParameter<int>))]
+        [InlineData(typeof(ClassWithSetter<int>))]
+        [InlineData(typeof(ClassWithCtorParameter<string>))]
+        [InlineData(typeof(ClassWithSetter<string>))]
+        [InlineData(typeof(FileInfo))]
+        [InlineData(typeof(FileInfo[]))]
+        [InlineData(typeof(string[]))]
+        [InlineData(typeof(List<string>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(List<int>))]
+        public async Task Handler_constructor_receives_option_arguments_bound_to_the_specified_type(
+            Type type)
+        {
+            var testCase = common.BindingCases[type];
+            ICommandHandler handler = CommandHandler.Create(
+                                                MakeGenericType(typeof(TesttCommandHandler<>), testCase.ParameterType)
+                                                    .GetMethod(nameof(TesttCommandHandler<int>.Invoke)));
+            Command command = GetSingleArgumentCommand(testCase);
+            command.Handler = handler;
+
+            var parseResult = command.Parse($"--value {testCase.CommandLine}");
+            var invocationContext = new InvocationContext(parseResult);
+            await handler.InvokeAsync(invocationContext);
+
+            var boundValue = ((BoundValueCapturer)invocationContext.InvocationResult).BoundValue;
+            boundValue.Should().BeAssignableTo(testCase.ParameterType);
+            testCase.AssertBoundValue(boundValue);
+        }
+
+        [Theory]
+        [InlineData(typeof(ClassWithCtorParameter<int>))]
+        [InlineData(typeof(ClassWithSetter<int>))]
+        [InlineData(typeof(ClassWithCtorParameter<string>))]
+        [InlineData(typeof(ClassWithSetter<string>))]
+        [InlineData(typeof(FileInfo))]
+        [InlineData(typeof(FileInfo[]))]
+        [InlineData(typeof(string[]))]
+        [InlineData(typeof(List<string>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(List<int>))]
+        public void Model_constructor_receives_option_arguments_bound_to_the_specified_type(
+                    Type type)
+        {
+            var testCase = common.BindingCases[type];
+            var typeToCreate = MakeGenericType(typeof(TestModel<>), testCase.ParameterType);
+            Command command = GetSingleArgumentCommand(testCase);
+
+            var binder = new ModelBinder(typeToCreate);
+            var commandLine = $"--value {testCase.CommandLine}";
+            var bindingContext = new BindingContext(command.Parse(commandLine));
+            var instance = binder.CreateInstance(bindingContext) as TestModelBase;
+
+            instance.Value.Should().BeAssignableTo(testCase.ParameterType);
+            testCase.AssertBoundValue(instance.Value);
+        }
+
+        private static Command GetSingleArgumentCommand(BindingTestCase testCase)
```

Comment:
This name could be a bit more accurate, since the argument we're targeting here is actually on an option, not a command.